taipy/gui_core/_context.py

```diff
@@ -1104,3 +1105,8 @@ def on_dag_select(self, state: State, id: str, payload: t.Dict[str, str]):
                     _warn(f"dag.on_action(): Exception raised in '{args[1]}()' with '{args[0]}'", e)
         elif args[1]:
             _warn(f"dag.on_action(): Invalid function '{args[1]}()'.")
+
+    def get_creation_reason(self):
+        if not (reason := can_create()):
```

Comment:
def get_creation_reason(self):
```suggestion
        f"Cannot create scenario: {reason.reasons}" if (reason := can_create()) else ""
```
?

Reply:
it would be f"Cannot ..." if not (reason :! ...)
I suppose it's more pythonic ...